File: convert-csv.js
const fs = require('fs');

// Read the CSV file
const fileContent = fs.readFileSync('./tmnf.csv', 'utf-8');
const lines = fileContent.split('\n');

// Parse CSV manually
const entries = [];
for (let i = 2; i < lines.length; i++) {
  const line = lines[i].trim();
  if (!line) break;
  
  // Split by comma, but be careful with quoted fields
  const parts = [];
  let current = '';
  let inQuotes = false;
  
  for (let j = 0; j < line.length; j++) {
    if (line[j] === '"') {
      inQuotes = !inQuotes;
    } else if (line[j] === ',' && !inQuotes) {
      parts.push(current.trim());
      current = '';
    } else {
      current += line[j];
    }
  }
  parts.push(current.trim());
  
  const track = parts[2]?.trim();
  const time = parts[3]?.trim();
  const vsRta = parts[4]?.trim();
  const authors = parts[5]?.trim();
  const date = parts[6]?.trim();
  
  // Stop at E05 (empty track means end)
  if (!track) break;
  
  // Skip if essential data is missing
  if (!time || !vsRta || !authors || !date) continue;
  
  entries.push({
    track,
    time,
    vsRta,
    authors: authors.split(' + ').map(a => a.trim()),
    date: convertDate(date),
    links: {
      video: '',
      replay: '',
      inputs: ''
    }
  });
}

// Convert date format from DD-Mon-YY to YYYY-MM-DD
function convertDate(dateStr) {
  const months = {
    'Jan': '01', 'Feb': '02', 'Mar': '03', 'Apr': '04', 'May': '05', 'Jun': '06',
    'Jul': '07', 'Aug': '08', 'Sep': '09', 'Oct': '10', 'Nov': '11', 'Dec': '12'
  };
  
  const parts = dateStr.split('-');
  if (parts.length !== 3) return dateStr;
  
  const day = parts[0].padStart(2, '0');
  const month = months[parts[1]];
  const year = '20' + parts[2];
  
  return `${year}-${month}-${day}`;
}

// Output TypeScript code
console.log('Copy this into leaderboards.ts as the entries for TMNF:');
console.log('\nentries: [');
entries.forEach((entry) => {
  console.log(`  {
    track: "${entry.track}",
    time: "${entry.time}",
    vsRta: "${entry.vsRta}",
    authors: [${entry.authors.map(a => `"${a}"`).join(', ')}],
    date: "${entry.date}",
    links: {
      video: "",
      replay: "",
      inputs: "",
    },
  },`);
});
console.log(']');

console.log(`\n\nTotal entries: ${entries.length}`);

File: convert-csv.py
import csv
from datetime import datetime

# Read CSV
with open('tmnf.csv', 'r', encoding='utf-8') as f:
    reader = csv.reader(f)
    lines = list(reader)

# Convert month
months = {
    'Jan': '01', 'Feb': '02', 'Mar': '03', 'Apr': '04', 'May': '05', 'Jun': '06',
    'Jul': '07', 'Aug': '08', 'Sep': '09', 'Oct': '10', 'Nov': '11', 'Dec': '12'
}

def convert_date(date_str):
    parts = date_str.split('-')
    if len(parts) != 3:
        return date_str
    day = parts[0].zfill(2)
    month = months.get(parts[1], '01')
    year = '20' + parts[2]
    return f"{year}-{month}-{day}"

# Parse entries
entries = []
for i in range(2, len(lines)):
    row = lines[i]
    if len(row) < 7 or not row[2].strip():
        break
    
    track = row[2].strip()
    time = row[3].strip()
    vsRta = row[4].strip()
    authors_str = row[5].strip()
    date = row[6].strip()
    
    if not all([track, time, vsRta, authors_str, date]):
        continue
    
    authors = [a.strip() for a in authors_str.split(' + ')]
    
    entries.append({
        'track': track,
        'time': time,
        'vsRta': vsRta,
        'authors': authors,
        'date': convert_date(date)
    })

# Output TypeScript
print("entries: [")
for entry in entries:
    authors_str = ", ".join(f'"{a}"' for a in entry['authors'])
    print(f'''  {{
    track: "{entry['track']}",
    time: "{entry['time']}",
    vsRta: "{entry['vsRta']}",
    authors: [{authors_str}],
    date: "{entry['date']}",
    links: {{
      video: "",
      replay: "",
      inputs: "",
    }},
  }},''')
print("]")
print(f"\n\nTotal entries: {len(entries)}")
